framework/framework.go

```diff
@@ -44,7 +44,7 @@ type Config struct {
 	// and different resources can be executed depending on the runtime object
 	// being reconciled.
 	ResourceRouter *ResourceRouter
-	K8sClient      kubernetes.Interface
+	RESTClient     rest.Interface
```

Comment:
This is actually a breaking change; we can't use `kubernetes.Interface` because the `restClient` GET we do does only work with a properly defined client. For discussion: https://gigantic.slack.com/archives/C2MS2VB37/p1523946503000135

Reply:
I would like to keep this as it was. Did you actually try `g8sClient.Discovery().RESTClient()`? The discovery one should be untyped.

Reply:
When this works some godoc should also be provided to explain what we need and why that is.

Reply:
Using `g8sClient.Discovery().RESTClient()` doesn't help here, that is also a `rest.Interface`. We need a common interface which works for standard k8s objects and CRs, `rest.Interface` seems to be the only one that works.

I would be happy if we can avoid this change, but I can't see a solution for now. I'll add some godoc.

Reply:
All `RESTClient()` calls give you `rest.Interface`. The problem is the encoding the rest clients do internally AFAIK. I am not into details though. How should anybody now know which kind of rest client interface to provide here? That does not work. It should be possible to accept some specific interface which does not cause the user to fail occasionally. What about accepting the discovery interface here? We get a rest client from it and it should be "abstracted" in that sense.

Reply:
I tried this, it doesn't work for either `NodeConfig` or `Configmaps`. So I don't even know what that client is even good for. This sucks.
The _only_ working client I found for CRs so far has been `g8sClient.ProviderV1alpha1().RESTClient()`. Nothing else I tried worked ...

Reply:
`NodeConfig` is actually `CoreV1Alpha1` so why does `ProviderV1Alpha1` then work? Can you come up with a tiny `main.go` where you can reproduce this? Lets break down the problem and look into this. Check https://github.com/giantswarm/apiextensions/tree/master/pkg/clientset/ what the generated clients are doing. They create a rest client with the codec magic. There is also `VersionedParams(&options, scheme.ParameterCodec)` in the client methods when making up requests. Maybe you also find somebody to talk to in the k8s Slack.